cephfs: remove nodeplugin RBAC for cephfs

humblec opened the pull request:
The cephfs nodeplugin RBAC is not required and this commit
remove the same.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@Mergifyio rebase

humblec commented:
@Madhu-1  ptal..  thanks,

humblec commented:
@Mergifyio rebase

humblec commented:
/retest ci/centos/mini-e2e-helm/k8s-1.22

Madhu-1 commented on deploy/cephfs/kubernetes/csi-nodeplugin-rbac.yaml:
```diff
@@ -9,9 +9,6 @@ apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: cephfs-csi-nodeplugin
 rules:
-  - apiGroups: [""]
```
Why keep empty ClusterRole without any rules?

humblec replied:
removed it

Madhu-1 reviewed (changes requested):
Need to take care in below places

* Need to remove from helm chart
* Doc update is required.
* Remove it from nfs plugin also.

humblec commented:
> Need to take care in below places
> 
>     * Need to remove from helm chart
> 
>     * Doc update is required.
> 
>     * Remove it from nfs plugin also.

This was kept for CephFS specifically, however I can update the NFS side too in the same PR.

humblec committed:
cephfs: remove nodeplugin RBAC

This commit remove the clusterRole and Binding of cephfs node plugin
as the node RBAC is not needed for CephFS.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
helm: remove cephfs node plugin cluster role RBAC

This commit remove the node plugin RBAC of CephFS.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec committed:
doc: remove the clusterRole sections from the upgrade guide

As we have removed the clusterrole and binding the subjected
documentation also removed.

Signed-off-by: Humble Chirammal <hchiramm@redhat.com>

humblec commented:
@Madhu-1 nfs plugin has not addressed here and will send it in a different PR.. rest of the comments are addressed. PTAL. Thanks!

Madhu-1 commented on charts/ceph-csi-cephfs/templates/nodeplugin-clusterrole.yaml:
```diff
@@ -1,18 +0,0 @@
-    release: {{ .Release.Name }}
-    heritage: {{ .Release.Service }}
-rules:
-  - apiGroups: [""]
```
clusterrole binding still needs to be removed?

humblec replied:
oh.. yeah.. removed it @Madhu-1

humblec commented:
/retest ci/centos/k8s-e2e-external-storage/1.23
